src/rrdp-iostat/rrdp_iostat.ml

```diff
@@ -28,12 +28,39 @@ let with_xc f = Xenctrl.with_intf f
 
 (* Return a list of (domid, uuid) pairs for domUs running on this host *)
 let get_running_domUs xc =
-  let open Xenctrl in
-  Xenctrl.domain_getinfolist xc 0 |> List.map (fun di ->
-      let domid = di.domid in
-      let uuid = Uuid.to_string (Uuid.uuid_of_int_array di.handle) in
-      (domid, uuid)
-    ) |> List.filter (fun x -> fst x <> 0)
+  let metadata_of_domain di =
+    let open Xenctrl in
+    let domid = di.domid in
+    let uuid = Uuid.(to_string (uuid_of_int_array di.handle)) in
+
+    (* Actively hide migrating VM uuids, these are temporary and xenops
+       writes the original and the final uuid to xenstore *)
+    let uuid_from_key key =
+      let path = Printf.sprintf "/vm/%s/%s" uuid key in
+      try
+        Xenstore.(with_xs (fun xs -> xs.read path))
+      with Xs_protocol.Enoent _hint ->
+        D.info "Couldn't read path %s; falling back to actual uuid" path;
+        uuid
+    in
+    let stable_uuid = Option.fold ~none:uuid ~some:uuid_from_key in
+
+    (* Do not list dom0 *)
+    if domid = 0 then
+      None
+    else
+      let key =
+        if Astring.String.is_suffix ~affix:"000000000000" uuid then
+          Some "origin-uuid"
+        else if Astring.String.is_suffix ~affix:"000000000001" uuid then
+          Some "final-uuid"
+        else
+          None
+      in
+      Some (domid, stable_uuid key)
+  in
+  Xenctrl.domain_getinfolist xc 0
```

Comment:
should this `0` be `domid`?

Reply:
I think it should be `1`, this way I don't have to filter out domain 0 afterwards.

```
let domain_getinfolist handle first_domain =
```